src/test/client/zerocopy_sanity.cc

```diff
@@ -20,20 +20,27 @@
 
 // the only unittest we can do is to test the comibnation of
 // invalid parameters in libcephfs functions that can use zerocopy
+// ceph_ll_readv_writev and ceph_ll_nonblocking_readv_writev doesn't support
+// zerocopy, only the _v2 API does
 TEST(libcephfs, zerocopy_sanity_invalid_params) {
-  struct ceph_mount_info *mount_info = nullptr;
-  auto res = ceph_create(&mount_info, "test_mount");
+  struct ceph_mount_info *cmount = nullptr;
+  auto res = ceph_create(&cmount, NULL);
   ASSERT_EQ(0, res);
-  struct ceph_ll_io_info io_info;
-  struct iovec iov;
-  io_info.iov = &iov;
-  io_info.iovcnt = 0;
-  io_info.write = false;
+  ceph_ll_io_info_v2 io_info;
+  iovec iov;
+  io_info.base_info.iov = &iov;
+  io_info.base_info.iovcnt = 0;
+  io_info.base_info.write = false;
   io_info.zerocopy = true;
+  res = ceph_ll_readv_writev_v2(cmount, &io_info.base_info, sizeof(ceph_ll_io_info_v2));
```

Comment:
How is this returning -EINVAL? info_size is sizeof(v2) struct and  the check returns -EINVAL if

> if (!info || info_size < sizeof(ceph_ll_io_info)) {

Reply:
Ah, this line asserts the zerocopy params that are validates in the client:
https://github.com/ceph/ceph/blob/458a032e2e1accb3119d3d59bb7de31cdf99f8fd/src/client/Client.cc#L11520
The code you mentioned is asserted here
https://github.com/ceph/ceph/blob/458a032e2e1accb3119d3d59bb7de31cdf99f8fd/src/test/client/zerocopy_sanity.cc#L41
where the size is set to be less then required minimum